packages/jest-jasmine2/src/index.ts

```diff
@@ -47,38 +47,30 @@ async function jasmine2(
   // TODO: Remove config option if V8 exposes some way of getting location of caller
   // in a future version
   if (config.testLocationInResults === true) {
-    const originalIt = environment.global.it;
-    environment.global.it = ((...args) => {
-      const stack = getCallsite(1, runtime.getSourceMaps());
-      const it = originalIt(...args);
-
-      // @ts-expect-error
-      it.result.__callsite = stack;
-
-      return it;
-    }) as Global.Global['it'];
-
-    const originalXit = environment.global.xit;
-    environment.global.xit = ((...args) => {
-      const stack = getCallsite(1, runtime.getSourceMaps());
-      const xit = originalXit(...args);
-
-      // @ts-expect-error
-      xit.result.__callsite = stack;
-
-      return xit;
-    }) as Global.Global['xit'];
-
-    const originalFit = environment.global.fit;
-    environment.global.fit = ((...args) => {
-      const stack = getCallsite(1, runtime.getSourceMaps());
-      const fit = originalFit(...args);
-
-      // @ts-expect-error
-      fit.result.__callsite = stack;
+    function wrapIt<T extends Global.ItBase>(original: T): T {
+      const wrapped = (
+        testName: Global.TestName,
+        fn: Global.TestFn,
+        timeout?: number,
+      ) => {
```

Comment:
```suggestion
      const wrapped: Global.ItBase = (
        testName,
        fn,
        timeout,
      ) => {
```

or something like that?

Reply:
Not possible, for the same reason as I have to cast to `any` below - the `ItBase` type has `.each` as a property that's not given here.
<img width="767" alt="Screen Shot 2020-08-17 at 07 58 13" src="https://user-images.githubusercontent.com/329222/90361815-6b018c80-e05f-11ea-870d-8a9e7d316f15.png">